### ISSUE TITLE
Access VPN in builds

### ISSUE THREAD

**Issue by mei-li** [Shippable/support#4457]
### Description of your issue:
We want to access our VPN in our shippable builds, to access some resources and eventually deploy. We have a paid subscription plan. Is this possible, what are our options. Our shippable.yml looks a bit like that.
```
build:
    pre_ci_boot:
        image_name: ...
        image_tag: IMG_TAG
    ci:```

**Comment by trriplejay** [comment]
One option would be to utilize the [BYON feature](http://docs.shippable.com/platform/tutorial/runtime/manage-byon-nodes/) to attach machines to your subscription that are already in your private network.  This way, there would be no special permission or connection required, since it would be your own machine that is already part of your network.

What kind of resources are you trying to access? If it's a private docker image on your own registry, for example, then we offer a [docker registry integration](http://docs.shippable.com/platform/integration/dockerRegistryLogin/) where you can supply the credentials in a secure integration, and then during the CI step we will automatically log in so that you can pull/push.

Basically if any of your private resources are restricted via some kind of credential, you can probably add an integration to grant yourself access from one of the on-demand machines, but if it's easier, you could also just use the BYON feature as I described above.

Let me know if you have any other questions.

**Comment by mei-li** [comment]
We already have docker registry integration. It seems after some discussion we can reduce our needs and only need an ssh tunnel to forward a couple of port to access some solr indexes. Is there a way to do this securely with the on demand machines?

**Comment by trriplejay** [comment]
Perhaps the [ssh key integration](http://docs.shippable.com/platform/integration/sshKey/) would meet your requirements? You can generate a new sshkey, copy the public key to whatever machine you want to tunnel into, and if you include the integration in your build, the private key will automatically be available on the build machine.

**Comment by mei-li** [comment]
Thank you

**Comment by manishas** [comment]
Closing since the question was answered.
